src/app/new-courses-page/new-course-form/new-course-form.component.ts

```diff
@@ -4,38 +4,167 @@ import {
+  courseForm: FormGroup;
+  selectedAuthors: Option[];
+  validationMessage: CourseErrors = {
+    name: '',
+    description: '',
+    date: '',
+    length: '',
+    authors: ''
+  };
+
+  get name(): AbstractControl {
+    return this.courseForm.get('name');
+  }
+
+  get description(): AbstractControl {
+    return this.courseForm.get('description');
+  }
+
+  get date(): AbstractControl {
+    return this.courseForm.get('date');
+  }
+
+  get length(): AbstractControl {
+    return this.courseForm.get('length');
+  }
+
+  get authors() {
+    return this.courseForm.get('authors') as FormArray;
+  }
+
+  private validationMessagesMap = {
+    name: {
+      required: '*Please enter title',
+      maxlength: '*Max length 50 symbols'
+    },
+    description: {
+      required: '*Please enter description',      
+      maxlength: '*Max length 500 symbols'
+    },
+    date: {
+      required: '*Please enter date',      
+      pattern: '*Doesn\'t feet pattern dd/mm/yyyy'
+    },
+    length: {
+      required: '*Please enter duration',      
+      pattern: '*Only digits can be used'
+    },
+    authors: {
+      required: '*Please enter authors'
+    }
+  };
+
+  constructor(
+    private fb: FormBuilder,
+  ) {}
+
+  ngOnInit(): void {
+    this.buildForm();
+  }
 
   ngOnChanges(): void {
-    if (this.course) {
-      this.value = this.course;
+    if (this.course && this.first) {
+      this.name.setValue(this.course.name);
+      this.description.setValue(this.course.description);
+      this.date.setValue(this.course.date);
+      this.length.setValue(this.course.length);
+      this.authors.setValue(this.course.authors);
       this.courseTitle = "Edit course";
-    }    
+      this.first = false;
+    }
+  }
+
+  onBlur(event: FocusEvent) {
+    if ((event.target as Element).id === 'name') {
+      this.setValidationMessage(this.name, 'name');
+    } else if ((event.target as Element).id === 'description') {
+      this.setValidationMessage(this.description, 'description');
+    }
+  }
+
+  onClick(controlName: string) {
+    if (controlName === 'date') {
+      this.setValidationMessage(this.date, 'date');
+    } else if (controlName === 'length') {
+      this.setValidationMessage(this.length, 'length');
+    } else if (controlName === 'authors') {
+      this.setValidationMessage(this.authors, 'authors');
+    }
+  }
+
+  onReceiveCheckedAuthors(authors: Option[]) {
+    if (authors && this.courseForm.valid) {
+      this.authors.setValue([]);
+      this.newCourse.emit(
+        this.course 
+          ? {...this.courseForm.value, id: this.course.id, authors: authors}
+          : {...this.courseForm.value, authors: authors}
+        )
+    }
   }
 
   onSave(): void {
-    this.newCourse.emit(this.value);
+    this.showFormErrors = true;
   }
 
   onCancel(): void {
     this.cancel.emit();
   }
+
+  setValidationMessage(c: AbstractControl, controlName: string) {
+    for (const field in this.validationMessage) {
```

Comment:
try to avoid such dynamic code, it's hard to debug and sometimes may result in a security bridge